### DISCUSSION TITLE
How to set entity prefix?

### DISCUSSION THREAD

**Discussion by Lenbok** [alandtse/tesla#273] [Q&A]
I have installed the integration and it is connected to my tesla happily. However the entity names do not have any prefix like I have seen when other people post their entity ids. For example, the charger sensor is named `binary_sensor.charger_sensor`. 

How can I make all the entity ids a bit less generic and incorporate the name of the car? I tried renaming the device and HA said it would update all the entity ids, but it actually did not.

**Comment by mekaneck** [comment]
I know this is old, but I have the exact same issue. I'd rather not go through and rename every single entity, but it appears I might have to do that. I'm curious if something is different with my installation or if this is normally how it works? Does everyone really have, for example, a `lock.doors` entity like I do, or is it by default named something more specific like `lock.tesla_model_3_doors`?

**Reply by alandtse** [reply]
This happens when you don't have a name set up in Tesla for the car. That said, if renaming the entity on the device fails, that may just be a HA bug since we don't control renaming at this level.

**Reply by mekaneck** [reply]
Thank you! This was exactly the problem. I named my car (from the display in the car itself), removed the HA integration, and then re-added it, and the prefix is now there in all the entities. And renaming the device in HA now renames all entities as expected. So I agree, I think there is an HA bug where if an entity prefix isn't initially provided, then renaming the device has no effect. I'll do some digging and may create an issue over there.
